feat(manager/sbt): Improve scala 3 dependencies handling and meta-build classes

ArtyomGabeev opened the pull request:
<!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
<!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->
<!-- Please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md before you create your pull request.-->

## Changes

1. Parse build.sbt first and pass scalaVersion to remaining files if defined.

<!-- Describe what behavior is changed by this PR. -->

## Context

Fixes:
https://github.com/renovatebot/renovate/issues/29127

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

<!-- Do you have any suggestions about this PR template? Edit it here: https://github.com/renovatebot/renovate/edit/main/.github/pull_request_template.md -->

<!-- Please do not force push to your PR's branch after you have created your PR, as doing so forces us to review the whole PR again. This makes it harder for us to review your work because we don't know what has changed. -->
<!-- PRs will always be squashed by us when we merge your work. Commit as many times as you need in this branch. -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/renovatebot/renovate?pullRequest=29155) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/renovatebot/renovate?pullRequest=29155) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/renovatebot/renovate?pullRequest=29155) it.</sub>

ArtyomGabeev commented:
Yes, Ill rebase and clean up pr, it is already covered in mentioned pr

ArtyomGabeev commented:
> Is there any overlap/conflict with #29291 which needs to be considered?

I've rebased, performed clean-up, tested locally and updated description.
